src/installer/corehost/coreclr_resolver_t.h

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __CORECLR_RESOLVER_T_H__
+#define __CORECLR_RESOLVER_T_H__
+
+#include "pal.h"
+#include "error_codes.h"
+
+using host_handle_t = void*;
+
+// Prototype of the coreclr_initialize function from coreclr.dll
+using coreclr_initialize_fn = pal::hresult_t(STDMETHODCALLTYPE*)(
+    const char* exePath,
+    const char* appDomainFriendlyName,
+    int propertyCount,
+    const char** propertyKeys,
+    const char** propertyValues,
+    host_handle_t* hostHandle,
+    unsigned int* domainId);
+
+// Prototype of the coreclr_shutdown function from coreclr.dll
+using coreclr_shutdown_fn = pal::hresult_t(STDMETHODCALLTYPE*)(
+    host_handle_t hostHandle,
+    unsigned int domainId,
+    int* latchedExitCode);
+
+// Prototype of the coreclr_execute_assembly function from coreclr.dll
+using coreclr_execute_assembly_fn = pal::hresult_t(STDMETHODCALLTYPE*)(
+    host_handle_t hostHandle,
+    unsigned int domainId,
+    int argc,
+    const char** argv,
+    const char* managedAssemblyPath,
+    unsigned int* exitCode);
+
+// Prototype of the coreclr_create_delegate function from coreclr.dll
+using coreclr_create_delegate_fn = pal::hresult_t(STDMETHODCALLTYPE*)(
+    host_handle_t hostHandle,
+    unsigned int domainId,
+    const char* entryPointAssemblyName,
+    const char* entryPointTypeName,
+    const char* entryPointMethodName,
+    void** delegate);
+
+struct coreclr_resolver_contract_t
```

Comment:
I would store the coreclr library handle in the contract as well - just in case...